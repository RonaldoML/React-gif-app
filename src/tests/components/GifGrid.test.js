import React, { PureComponent } from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Prueba de GifGrid', () => {

    const category = 'Lol';

    test('should mostrar correctamente el componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} ></GifGrid>)
        expect(wrapper).toMatchSnapshot();

    });

    test('should mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cualquiercosa.jpg',
                title: 'Cualquier cosa'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        //Hacer como si el componente ha cargado informacion
        const wrapper = shallow(<GifGrid category={category} ></GifGrid>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(true);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });


});
