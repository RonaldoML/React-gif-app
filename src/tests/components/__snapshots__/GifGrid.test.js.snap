// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGrid should mostrar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animated__fadeIn"
  >
    Lol
  </h3>
  <p
    className="animate__animated animated__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba de GifGrid should mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animated__fadeIn"
  >
    Lol
  </h3>
  <p
    className="animate__animated animated__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
